JustSaying.IntegrationTests/Fluent/Publishing/WhenAMessageIsPublishedToAQueue.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using System.Threading.Tasks;
+using JustSaying.Messaging;
+using JustSaying.TestingFramework;
+using Microsoft.Extensions.DependencyInjection;
+using NSubstitute;
+using Xunit.Abstractions;
+
+namespace JustSaying.IntegrationTests.Fluent.Publishing
+{
+    public class WhenAMessageIsPublishedToAQueue : IntegrationTestBase
+    {
+        public WhenAMessageIsPublishedToAQueue(ITestOutputHelper outputHelper)
+            : base(outputHelper)
+        {
+        }
+
+        [AwsFact]
+        public async Task Then_The_Message_Is_Handled()
+        {
+            // Arrange
+            var completionSource = new TaskCompletionSource<object>();
+            var handler = CreateHandler<SimpleMessage>(completionSource);
+
+            var services = GivenJustSaying()
+                .ConfigureJustSaying((builder) => builder.Publications((options) => options.WithQueue<SimpleMessage>(UniqueName)))
+                .ConfigureJustSaying((builder) => builder.Subscriptions((options) => options.ForQueue<SimpleMessage>(UniqueName)))
+                .AddSingleton(handler);
+
+            string content = Guid.NewGuid().ToString();
+
+            var message = new SimpleMessage()
+            {
+                Content = content
+            };
+
+            await WhenAsync(
+                services,
+                async (publisher, listener, cancellationToken) =>
```

Comment:
Stylistic, but when you have a long complex lambda, I prefer to extract it as a method with a descriptive name. In this case it would be something like `PublishMessageAndWaitForReceipt` ?

Reply:
That then doesn't really flow as a simple all-in-one test method you just read and see what it's doing, as then you've got the _Arrange_ in the test method, and the _Act_ and _Assert_ off somewhere else in another method that the test method calls.

Happy to refactor further, but I think there readability is more important over "efficient engineering"

Reply:
I find the named method more readable.

Reply:
YMMV 🤷‍♂️